### ISSUE TITLE
GptService::GenerateTests creates invalid test files

### ISSUE THREAD

**Issue by MrSerth** [openHPI/codeharbor#1576]
The `GptService::GenerateTests` currently creates test files as follows:

https://github.com/openHPI/codeharbor/blob/992aeab7e085e286b48cf4d5710e467617af2fd3/app/services/gpt_service/generate_tests.rb#L16

However, the option `visible: false` is invalid, since `visible` is a string (no boolean) with fixed values (according to the ProFormA standard). When saved in the database, the boolean `false` is saved as `f`. 

When downloading such a task as ProFormA ZIP file, the following error is raised:

```ruby
ProformaXML::PostGenerateValidationError at /tasks/59/download
[#<Nokogiri::XML::SyntaxError: 281:0: ERROR: Element '{urn:proforma:v2.1}file', attribute 'visible': [facet 'enumeration'] The value 'f' is not an element of the set {'yes', 'no', 'delayed'}.>]
```

We should:

1. fix the error in the unit test generation
2. add a validation to the filed
3. test the behavior
4. (maybe) write a migration to resolve invalid values in our production database

**Comment by Melhaya** [comment]
@MrSerth Thanks for identifying the issue.

I have added a validation in the `TaskFile` and adjusted the creation of one in the `GptService`. 

Currently a few tests are failing though. I am not sure how to tackle this. Of course due to the added validation in the `TaskFile`, these errors are raised.  I am just not sure how to fix them. Could you please advise? thanks.

**Comment by MrSerth** [comment]
Fixed by #1588.
